### PR TITLE
Publish the artifact in SonaType or other Maven artifactory #5

### PR DESCRIPTION
**Pull request by BartekGravity**
* Provide a deployment pipeline
    * Prepare the artifact for SonaType release

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -14,28 +14,37 @@ on:
   pull_request:
     branches: [ "main" ]
 
+env:
+  MAVEN_USERNAME: ${{ secrets.TOKEN_USERNAME }}
```
Comment (MateuszGravity9) [inline .github/workflows/maven.yml:18]: maybe NEXUS_USERNAME  and NEXUS_PASSWORD?

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -14,28 +14,37 @@ on:
...
+  MAVEN_PASSWORD: ${{ secrets.TOKEN_PASSWORD }}
+  GPG_PASSPHRASE: ${{ secrets.SIGNING_PASSWORD }}
```
Comment (MateuszGravity9) [inline .github/workflows/maven.yml:20]: SIGNING_KEY_PASSPHRASE ?

File: .github/workflows/maven.yml [hunk 3]
```diff
@@ -14,28 +14,37 @@ on:
...
   build:
-
     runs-on: ubuntu-latest
-
     steps:
+    - name: Configure GPG Key
+      run: |
+        echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import --batch
+      env:
+        GPG_SIGNING_KEY: ${{ secrets.SIGNING_KEY_BASE64 }}
```
Comment (MateuszGravity9) [inline .github/workflows/maven.yml:30]: SIGNING_KEY?

**Review by MateuszGravity9 — COMMENTED** [review]
Minor naming changes - just a proposal, it seems more readable to me that way but I am not insisting

**Review by BartekGravity — COMMENTED** [review]
LGTM :)

File: .github/workflows/maven.yml (hunk 2 above)
Comment (jbibro) [reply .github/workflows/maven.yml]: done

File: .github/workflows/maven.yml (hunk 3 above)
Comment (jbibro) [reply .github/workflows/maven.yml]: done

File: .github/workflows/maven.yml (hunk 1 above)
Comment (jbibro) [reply .github/workflows/maven.yml]: done

**Review by MateuszGravity9 — APPROVED** [review]
lgtm

**Pull request merged**